src/schemas/Program/index.js

```diff
@@ -303,10 +316,21 @@ const resolvers = {
 	},
 	Query: {
 		program: async (obj, args, context, info) => {
+			const requestedFields = info.fieldNodes.flatMap((fieldNode) =>
+				fieldNode.selectionSet.selections.map((selection) => selection.name.value),
+			);
 			const { egoToken } = context;
 			const { shortName } = args;
-			return resolveSingleProgram(egoToken, shortName);
+
+			//create a condition to determine using either public or private endpoint
+			const hasPrivateField = requestedFields.filter((requestedField) =>
+				programServicePrivateFields.indexOf(requestedField) != -1 ? true : false,
+			);
+			return hasPrivateField.length > 0
+				? resolveSingleProgram(egoToken, shortName)
+				: resolveHTTPProgram(shortName);
```

Comment:
this works great - I think we can improve the syntax a little with some of the modern JS api methods.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

```javascript
const hasPrivateField = requestedFields.some((requestedField) => programServicePrivatesFields.includes(requestedField))

return hasPrivateField ? resolveSingleProgram(egoToken, shortName): resolveHTTPProgram(shortName);
```

Why is this better?
`filter` is most often used to filter down elements into a subset. But the only reason for the subset here if to check if there are any elements (`hasPrivateField.length > 0`)

`some` will also early return, so if a condition is met on the 1st element of an array, the condition is fulfilled,instead of iterating the entire array. This works because `hasPrivateField.length > 0` we're not checking if `hasPrivateField.length  = 4` or `hasPrivateField.length < 12`, it just needs to be greater than zero.

Why not use `indexOf`?
we didn't always have an `Array.includes` function, so people would check if an index exists, then return a boolean based on that. Thankfully we do have `Array.includes` now! 
use `includes` when you want a boolean, use `indexOf` when you need to use the actual index of the element.

The nice thing is we don't need the ternary anymore too. Whenever you see a ternary that is `condition ? true : false` there's often a function that will return a boolean.